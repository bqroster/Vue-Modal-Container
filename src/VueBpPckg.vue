<template>
    <div class="vue-bp-pckg-wrapper">
        <span>{{ getTitle }}</span>
    </div>
</template>

<script>

    export default {
        props: {
            title: {
                type: String,
                default: 'This is a simple BP Vue NPM Package'
            }
        },
        computed: {
            getTitle() {
                return this.title.trim().toLowerCase();
            }
        }
    }

</script>