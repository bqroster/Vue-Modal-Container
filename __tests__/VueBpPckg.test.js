import { mount } from '@vue/test-utils';
import VueBpPckg from "~VueBpPckg";

describe('VueBpPckg Component mounted', () => {

    test('VueBpPckg mounted correctly', () => {
        const wrapper = mount(VueBpPckg);
        expect(wrapper.element).toMatchSnapshot();
    });

    test('VueBpPckg with title default property', () => {
        const wrapper = mount(VueBpPckg);
        expect(wrapper.find('.vue-bp-pckg-wrapper').exists()).toBe(true);
        expect(wrapper.get('.vue-bp-pckg-wrapper').text()).toBe('this is a simple bp vue npm package');
    });

    test('VueBpPckg with title prop set', () => {
        const wrapper = mount(VueBpPckg,
            {
                propsData: {
                    title: 'This is a package'
                }
            });
        expect(wrapper.element).toMatchSnapshot();
        expect(wrapper.find('.vue-bp-pckg-wrapper').exists()).toBe(true);
        expect(wrapper.get('.vue-bp-pckg-wrapper').text()).toBe('this is a package');
    });
});