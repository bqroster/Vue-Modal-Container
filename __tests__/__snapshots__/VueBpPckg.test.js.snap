// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueBpPckg Component mounted VueBpPckg mounted correctly 1`] = `
<div
  class="vue-bp-pckg-wrapper"
>
  <span>
    this is a simple bp vue npm package
  </span>
</div>
`;

exports[`VueBpPckg Component mounted VueBpPckg with title prop set 1`] = `
<div
  class="vue-bp-pckg-wrapper"
>
  <span>
    this is a package
  </span>
</div>
`;
